numpy/random/generator.pyx

```diff
@@ -3950,13 +3954,19 @@ cdef class Generator:
                [0, 1, 2],
                [3, 4, 5]])
 
+        >>> rng.permutation("abc")
+        Traceback (most recent call last):
+            ...
+        numpy.AxisError: x must be an integer or at least 1-dimensional
         """
-        if isinstance(x, (int, np.integer)):
+        arr = np.asarray(x)
+        if arr.ndim < 1 and arr.dtype.kind == 'i':
```

Comment:
I do not mind raising the error personally for `np.array(1)`. The old check seems OK to me?